plotly/tests/test_core/test_plotly/test_plot.py

```diff
@@ -66,3 +66,99 @@ def test_plot_empty_data(self):
         py.sign_in('PlotlyImageTest', '786r5mecv0')
         self.assertRaises(PlotlyEmptyDataError, py.plot, [],
                           filename='plot_invalid')
+
+    def test_plot_sharing_argument(self):
+
+        # Raise an error if sharing argument is incorrect
+        # correct arguments {'public, 'private', 'secret'}
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+
+        kwargs = {'filename': 'invalid-sharing-argument',
+                  'sharing': 'privste'}
+        self.assertRaises(PlotlyError, py.plot, fig, **kwargs)
```

Comment:
:tiger2: This can be stronger/more-readable if you use [`self.assertRaisesRegexp`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaisesRegexp) by opening a *context*

```
with self.assertRaisesRegexp(PlotlyError, 'the-exact-message-would-work'):
    py.plot(fig, filename='invalid-sharing-argument', sharing='privste')
```

If you feel comfy with regex's, you can even make it flexible by just making sure the keyword 'sharing' is in there.